scvi/model/base/_pyromixin.py

```diff
@@ -0,0 +1,415 @@
+import logging
+from typing import Optional, Union
+
+import numpy as np
+import torch
+from pyro import poutine
+from pytorch_lightning.callbacks import Callback
+
+from scvi.dataloaders import AnnDataLoader, DataSplitter, DeviceBackedDataSplitter
+from scvi.model._utils import parse_use_gpu_arg
+from scvi.train import PyroTrainingPlan, TrainRunner
+from scvi.utils import track
+
+logger = logging.getLogger(__name__)
+
+Number = Union[int, float]
+
+
+class PyroJitGuideWarmup(Callback):
+    def __init__(self, train_dl) -> None:
+        super().__init__()
+        self.dl = train_dl
+
+    def on_train_start(self, trainer, pl_module):
+        """
+        Way to warmup Pyro Guide in an automated way.
+        Also device agnostic.
```

Comment:
needs to be empty line above this one
```suggestion

        Also device agnostic.
```